## Loops in Python. ##


##### 1. Print first 10 natural numbers using while loop

In [2]:
x = 0

while x < 10:
    x +=1
    print(x)

1
2
3
4
5
6
7
8
9
10


##### 2. Print: 
    1 
    1 2 
    1 2 3 
    1 2 3 4 
    1 2 3 4 5

In [8]:
number_list = ''
x = 0

while x < 5:
    x += 1
    number_list += str(x)
    print(number_list)

    

1
12
123
1234
12345


In [18]:
for x in range(1,5+1):
    for j in range(1,x+1):
        print(j,end='')
    print()

1
12
123
1234
12345


##### 3. Calculate sum of all numbers from 1 to a given number

In [4]:
given = int(input("Please, enter a number"))
counter = 0
sum = 0

while counter <= given:
    sum += counter
    counter += 1

print(sum)

55


##### 4. Print multiplication table of a given number

In [13]:
given = int(input("Please, enter a number between 0 and 10"))

for x in range(0,11):
    result = given * x
    print(result)


0
5
10
15
20
25
30
35
40
45
50


##### 5. Print from a list all the number that are divisible by 5, smaller than 150. The loop should stop if an encountered number is greater than 500.

In [18]:
numbers = [12, 75, 150, 180, 145, 525, 50]

for x in numbers:
    if x > 500:
        break
    elif (x <= 150) and (x % 5 == 0) :
        print(x)
    else:
        continue

75
150
145


##### 6. Use a while loop to count the number of digits in a number.

In [ ]:
number = 7472956390
counter = 0

while number > 0:
    number = number // 10
    counter += 1

print(counter)

10


##### 7. Print:
    5 4 3 2 1
    4 3 2 1
    3 2 1
    2 1
    1
    


In [ ]:
number = 5

for x in range(number, 0, -1):
    for j in range(x, 0 ,-1):
        print(j,end='')
    print() 




54321
4321
321
21
1


##### 8. Print a list in reverse order using a loop.


In [41]:
list_direct = [10, 20, 30, 40, 50]
lis_inversed =[]

for x in range(len(list_direct), 0, -1):
    print(list_direct[x-1])
    


50
40
30
20
10


##### 9. Display numbers from -10 to -1 using for loop.

In [42]:
for num in range(-10,0):
    print(num)

-10
-9
-8
-7
-6
-5
-4
-3
-2
-1


##### 10. Print all prime numbers within a range.

In [53]:
left_limit = int(input("Enter a number, please."))
right_limit = int(input("Enter the second number."))

for number in range(left_limit, right_limit):
    if number > 1:
        for i in range(2, number):
            if (number % i == 0):
                break
        else:
            print(number)
            


29
31
37
41
43
47


##### 11. Print the Fibonacci sequence up to n terms.

In [7]:
terms = 10
number = 0
adding = 1

for x in range(0,terms):
    result = adding + number
    print(number, end=' ')
    number = adding
    adding = result 

0 1 1 2 3 5 8 13 21 34 

##### 12. Find the factorial of a given number using a loop.

In [6]:
n = 5
factorial = 1

for x in range(1,n+1):
    factorial = factorial * x    
print(factorial)

120


In [32]:
num_ = 5
factorial_ = 1

while num_ > 0 :
    factorial_ = factorial_ * num_
    num_ -= 1
print(factorial_)
    

120


##### 13. Given a number, return its reverse. For example, given 1234, return 4321.

In [5]:
given_n = 1234
reversed_n = 0

while given_n > 0:
    extracted_last_num = given_n % 10
    given_n = given_n // 10 

    #print(f"extracted    {extracted_last_num}")
    #print(f"still to be extract  {given_n}")

    reversed_n = (reversed_n * 10) + extracted_last_num
    
print(f"reversed    {reversed_n}")

reversed    4321


##### 14. From a list, print only the numbers in an even index using a loop.       

In [23]:
my_list = [10, 20, 30, 40, 50, 60, 70, 80, 90, 100]

for x in range(len(my_list)):
    if x % 2 == 0 and x != 0:
        print(f"Index: {x}, number: {my_list[x]}")


Index: 2, number: 30
Index: 4, number: 50
Index: 6, number: 70
Index: 8, number: 90


##### 15. Calculate the sum of the digits of a number using a loop.

In [26]:
number = int(input("Enter an integer number."))
digits_sum = 0

while number > 0:
    last_digit = number % 10    
    digits_sum += last_digit
    number = number // 10

print(digits_sum)


6


##### 16. Calculate the cube of the numbers from 1 to an input number.

In [28]:
input_num = int(input("Enter a number."))

for x in range(1,input_num+1):
    cube_num = x ** 3
    print(f"Number: {x}. The cube is: {cube_num}")

Number: 1. The cube is: 1
Number: 2. The cube is: 8
Number: 3. The cube is: 27
Number: 4. The cube is: 64


##### 17. Print the pattern:
    *
    * *
    * * *
    * * * *
    * * * * *
    * * * *
    * * *   
    * *
    *

In [12]:
limiter = 5


for x in range(1,(limiter*2)+1):
    if x < limiter:
        print('* '*x)
    elif x == limiter:
        print("* "*x)

for x in range((limiter-1),0,-1):       
    print("* "*x)
    
    

* 
* * 
* * * 
* * * * 
* * * * * 
* * * * 
* * * 
* * 
* 


##### 18. Print the multiplication table for a given number.

In [30]:
given = int(input("Please, enter a number"))
print(f"MULTIPLICATION TABLE OF {given}")
print("\r")
print(f"{'Multiplication':<5} {'Result':>10}")
print("\r")


for x in range(11):
    result = given * x
    print(f"{given} X {x:<5} {result :>12}")

MULTIPLICATION TABLE OF 4

Multiplication     Result

4 X 0                0
4 X 1                4
4 X 2                8
4 X 3               12
4 X 4               16
4 X 5               20
4 X 6               24
4 X 7               28
4 X 8               32
4 X 9               36
4 X 10              40


##### 19. Print the pattern:
    1
    2 3
    4 5 6
    7 8 9 10
    11 12 13 14 15

In [77]:
last_num = 5
init_num = 1
counter = 0

for x in range(1,last_num+1):
    string = ''
    for j in range(1,x+1):
        counter += 1
        string += f"{counter} "
    print(string)

1 
2 3 
4 5 6 
7 8 9 10 
11 12 13 14 15 


In [ ]:
counter = 0
number = 5
for i in range(1, 5+1):
    for j in rnage

    

1
2
3
4
5
